library/core/src/option.rs

```diff
@@ -1074,18 +1074,22 @@ impl<T> Option<T> {
         }
     }
 
-    /// Calls the provided closure with a reference to the contained value (if [`Some`]).
+    /// Calls `f` with a reference to the contained value if [`Some`] and
+    /// returns the original Option.
```

Comment:
This first sentence appears on its own in some contexts (at least in search results), without the signature with the argument name "f". So, it might be better not to refer to `f` here by name.

Reply:
Ah I was looking at the docs for Option::and_then which uses `f`, but I see most other methods use "a function". Changed to that.

Also I broke the "returns" statement into a separate paragraph.